test/test_cpp_extensions.py

```diff
@@ -106,6 +106,83 @@ def test_optional(self):
         has_value = cpp_extension.function_taking_optional(None)
         self.assertFalse(has_value)
 
+    def test_inline_jit_compile_extension(self):
+        cpp_source1 = '''
+        #include <torch/torch.h>
+        at::Tensor sin_add(at::Tensor x, at::Tensor y) {
+          return x.sin() + y.sin();
+        }
+        '''
+
+        cpp_source2 = '''
+        #include <torch/torch.h>
+        at::Tensor sin_add(at::Tensor x, at::Tensor y);
+        PYBIND11_MODULE(TORCH_EXTENSION_NAME, m) {
```

Comment:
A random idea: it could be possible to auto generate the `PYBIND11_MODULE` by letting the user specify a list of tuples (or dict) in python, that define the necessary arguments for pybind.

For example, if we have a list of dicts to be passed as a `signatures` keyword to `load_inline`, the user would be able to just call with  `signatures=[{'name': 'sin_add', 'doc': 'sin(a) + sin(b)'}]` and let `cname` be inferred from `name`.

What do you think?

Reply:
Yes definitely -- maybe even just a list of function names should work. I imagine often folks won't care about the docstring. So `signatures` should take a list or dictionary

Reply:
It's a bit tricky because we need to pick which `.cpp` source to put the `PYBIND11_MODULE` into. We'll just use the first one

Reply:
You could just write everything the user pass as a `.hpp` file, and import them all in a `main.cpp` that contains the `PYBIND11_MODULE`, right?

Reply:
True. Now that I think of it, I'm not sure it actually makes sense that `cpp_sources` and `cuda_sources` be a list of strings. I think it would be enough that there is one `cpp_source` and one` cuda_source`, what do you think?

Reply:
I think any of them is fine.
I could imagine people writing individual functions in strings, in which case having the list of sources makes sense.